.github/workflows/build-test.yml

```diff
@@ -190,51 +190,63 @@ jobs:
-        cache-key: cache-0-${{ runner.os }}-maturin
-        artifact-bin-name: maturin
-        artifact-upload-name: ${{ runner.os }}-maturin
-
   python-bindings:
-    needs: [preflight-check, static-code-analysis, cargo-nextest, maturin]
+    needs: preflight-check
+    # needs: [preflight-check, static-code-analysis, cargo-nextest]
     if: ${{ needs.changes.outputs.source-code == 'true' }}
     runs-on: ubuntu-latest
     steps:
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # ratchet:actions/checkout@v4
-     - name: Download artifact maturin
-       uses: ./.github/actions/download-cached-rust-tool
+
+     - name: Set up Python
+       uses: actions/setup-python@v5
        with:
-         artifact-bin-name: maturin
-         artifact-upload-name: ${{ runner.os }}-maturin
+         python-version: "3.12"
+
+     - name: Install Poetry
+       uses: snok/install-poetry@v1
+       with:
+         version: 2.1.3
+
+     - name: Install Dependencies
+       run: poetry --project iceoryx2-ffi/python install
 
      - name: Prepare Linux
        run: |
          internal/scripts/ci_prepare_ubuntu.sh
          uname -a
 
-     - uses: actions/setup-python@v5
-       with:
-         python-version: 3.x
+     - name: Run Linters
+       run: |
+         # Import Ordering
+         poetry --project iceoryx2-ffi/python run isort --check-only $(pwd)/examples/python
+         poetry --project iceoryx2-ffi/python run isort --check-only $(pwd)/iceoryx2-ffi/python/python-src/iceoryx2
+         poetry --project iceoryx2-ffi/python run isort --check-only $(pwd)/iceoryx2-ffi/python/tests
+
+         # Formatting
+         poetry --project iceoryx2-ffi/python run black --check $(pwd)/examples/python
+         poetry --project iceoryx2-ffi/python run black --check $(pwd)/iceoryx2-ffi/python/python-src/iceoryx2
+         poetry --project iceoryx2-ffi/python run black --check $(pwd)/iceoryx2-ffi/python/tests
 
-     - name: Language Binding Tests
-       run: internal/scripts/ci_check_python.sh check
+     - name: Build Python Bindings
+       run: |
+         poetry --project iceoryx2-ffi/python run maturin develop \
+                --manifest-path iceoryx2-ffi/python/Cargo.toml \
+                --target-dir target/ffi/python
+
+     - name: Test Python Bindings
+       run: |
+         poetry --project iceoryx2-ffi/python run pytest iceoryx2-ffi/python/tests/*
+
+     # NOTE: Bindings need to be built before running analysis
+     - name: Run Static Analysis
+       run: |
+         poetry --project iceoryx2-ffi/python run prospector -m -D -T -s veryhigh -F --profile .prospector-iceoryx2.yaml iceoryx2-ffi/python/python-src/iceoryx2
+         poetry --project iceoryx2-ffi/python run prospector -m -D -T -s veryhigh -F --profile .prospector-tests.yaml iceoryx2-ffi/python/tests
+         poetry --project iceoryx2-ffi/python run prospector -m -D -T -s veryhigh -F --profile .prospector.yaml examples/python
```

Comment:
Could this be added to a script?

It's always cumbersome with so many commands when the CI fails and one tries to reproduce it locally.

It's a bit off-topic but I think we also need to move some to the build and run steps into scripts, in order to make it easier to reproduce CI failures but we need to be careful to not have just one big script like in iceoryx classic. Like you already mentioned at a different thread, `just` would be neat for this.